controllers/k8ssandra/medusa_reconciler.go

```diff
@@ -0,0 +1,239 @@
+// Create or update volumes for medusa
+func updateMedusaVolumes(dcConfig *cassandra.DatacenterConfig, medusaSpec *medusaapi.MedusaClusterTemplate, logger logr.Logger) {
+	// Medusa config volume, containing medusa.ini
+	configVolumeIndex, found := findMedusaVolume(dcConfig.PodTemplateSpec, fmt.Sprintf("%s-medusa", dcConfig.Cluster))
+	configVolume := &corev1.Volume{
+		Name: fmt.Sprintf("%s-medusa", dcConfig.Cluster),
+		VolumeSource: corev1.VolumeSource{
+			ConfigMap: &corev1.ConfigMapVolumeSource{
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: fmt.Sprintf("%s-medusa", dcConfig.Cluster),
+				},
+			},
+		},
+	}
+
+	addOrUpdateVolume(dcConfig, configVolume, configVolumeIndex, found)
+
+	// Medusa credentials volume using the referenced secret
+	if medusaSpec.StorageProperties.StorageProvider != "local" {
+		// We're not using local storage for backups, which requires a secret with backend credentials
+		secretVolumeIndex, found := findMedusaVolume(dcConfig.PodTemplateSpec, medusaSpec.StorageProperties.StorageSecretRef)
+		secretVolume := &corev1.Volume{
+			Name: medusaSpec.StorageProperties.StorageSecretRef,
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: medusaSpec.StorageProperties.StorageSecretRef,
+				},
+			},
+		}
+
+		addOrUpdateVolume(dcConfig, secretVolume, secretVolumeIndex, found)
+	}
+
+	// Pod info volume
+	podInfoVolumeIndex, found := findMedusaVolume(dcConfig.PodTemplateSpec, "podinfo")
+	podInfoVolume := &corev1.Volume{
+		Name: "podinfo",
+		VolumeSource: corev1.VolumeSource{
+			DownwardAPI: &corev1.DownwardAPIVolumeSource{
+				Items: []corev1.DownwardAPIVolumeFile{
+					{
+						Path: "labels",
+						FieldRef: &corev1.ObjectFieldSelector{
+							FieldPath: "metadata.labels",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	addOrUpdateVolume(dcConfig, podInfoVolume, podInfoVolumeIndex, found)
+}
+
+func addOrUpdateVolume(dcConfig *cassandra.DatacenterConfig, volume *corev1.Volume, volumeIndex int, found bool) {
+	if !found {
+		// volume doesn't exist, we need to add it
+		dcConfig.PodTemplateSpec.Spec.Volumes = append(dcConfig.PodTemplateSpec.Spec.Volumes, *volume)
+	} else {
+		// Overwrite existing volume
+		dcConfig.PodTemplateSpec.Spec.Volumes[volumeIndex] = *volume
+	}
+}
+
+func findMedusaContainer(dcPodTemplateSpec *corev1.PodTemplateSpec, containerName string, initContainer bool) (int, bool) {
+	if dcPodTemplateSpec.Spec.InitContainers == nil {
+		dcPodTemplateSpec.Spec.InitContainers = []corev1.Container{}
+	}
+	if initContainer {
+		for i, container := range dcPodTemplateSpec.Spec.InitContainers {
+			if container.Name == containerName {
+				return i, true
+			}
+		}
+	} else {
+		for i, container := range dcPodTemplateSpec.Spec.Containers {
+			if container.Name == containerName {
+				return i, true
+			}
+		}
+	}
+
+	return -1, false
+}
+
+func findMedusaVolume(dcPodTemplateSpec *corev1.PodTemplateSpec, volumeName string) (int, bool) {
```

Comment:
Similar to my comment about the `FindMedusaContainer` function, what do you think about moving and renaming this to something like `FindVolume`?

Reply:
done 👍